src/main/java/com/sourcegraph/webhook/EventSerializer.java

```diff
@@ -36,6 +50,18 @@ private static JsonElement render(Object o) {
     private static Adapter<BuildStatusSetEvent> BuildStatusSetEventAdapter = ((element, event) -> {
         element.addProperty("commit", event.getCommitId());
         element.add("status", render(event.getBuildStatus()));
+
+        // Extract remote branches
+        PullRequestCommitSearchRequest searchRequest = new PullRequestCommitSearchRequest.Builder(event.getCommitId()).build();
+        PageRequest pageRequest = new PageRequestImpl(0,10);
```

Comment:
is it in some cases problematic just querying 10? (I assume that's what the 10 means, if it's something different nvm)

Reply:
I upped it to 100.. I'd be very surprised if a single commit was reference in more than 100 PRs :)

Reply:
Why not use a loop around it to make sure it doesn't lead to problems? See this snippet for how to build a `pager` out of the `pageRequest` and iterate over it:

https://github.com/sourcegraph/bitbucket-server-plugin/blob/6baff37073995900c93dc05a7b6ccac403e5d281/src/main/java/com/sourcegraph/permission/PermissionRouter.java#L81-L85

It's not a lot of code, it doesn't cost us anything if it does one iteration, but it saves future-us a big "oh no" in case we have a customer that has hundreds of PRs and rebases them all at the same time onto the same PR or something :)

Reply:
Thanks, I would never have figure this out:

```
PageProvider<ApplicationUser> pager = (pageRequest) -> userService.search(search, pageRequest); 
```